rasa_core_sdk/forms.py

```diff
@@ -173,8 +190,9 @@ def intent_is_desired(requested_slot_mapping, tracker):
         mapping_intents = requested_slot_mapping.get("intent", [])
         mapping_not_intents = requested_slot_mapping.get("not_intent", [])
         intent = tracker.latest_message.get("intent", {}).get("name")
-        return ((not mapping_intents and intent not in mapping_not_intents) or
-                intent in mapping_intents)
+        return (
```

Comment:
this needs some rewriting as well, e.g. an intermediate var storing the first part of the `or`?